### PR TITLE
feat(data): add configuration for API GraphQL Gen2 React Native E2E tests

### PR DESCRIPTION
**Pull request by david-mcafee**
#### Description of changes

feat(data): add configuration for API GraphQL Gen2 React Native E2E tests

[Sample app + E2E tests PR](https://github.com/aws-amplify/amplify-js-samples-staging/pull/816)

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/integ-config/detox-integ-all.yml [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
-  timeout_minutes: 120
-- test_name: 'integ_rn_ios_device_tracking'
-  working_directory: amplify-js-samples-staging/samples/react-native/auth/deviceTracking
-  timeout_minutes: 120
-- test_name: 'integ_rn_ios_datastore_sqlite_adapter'
-  working_directory: amplify-js-samples-staging/samples/react-native/datastore/SQLiteAdapter
-  timeout_minutes: 120
-- test_name: 'integ_rn_ios_api_v6_rn_72_detox_cli'
-  working_directory: amplify-js-samples-staging/samples/react-native/api/v6/ApiGRAPHQL
+# - test_name: 'integ_rn_ios_storage'
```
Comment (david-mcafee) [inline .github/integ-config/detox-integ-all.yml:1]: Updated for faster CI runs for testing. Will remove before merging PR.

File: .github/workflows/callable-e2e-tests.yml [hunk 2]
```diff
@@ -23,25 +23,25 @@ jobs:
...
-      test_name: ${{ matrix.integ-config.test_name }}
-      framework: ${{ matrix.integ-config.framework }}
-      category: ${{ matrix.integ-config.category }}
-      spec: ${{ matrix.integ-config.spec || '' }}
-      amplifyjs_dir: ${{ matrix.integ-config.amplifyjs_dir || false }}
-      sample_name: ${{ toJSON(matrix.integ-config.sample_name) || '[""]' }}
-      browser: ${{ toJSON(matrix.integ-config.browser) || '[""]' }}
-      timeout_minutes: ${{ matrix.integ-config.timeout_minutes || 35 }}
-      retry_count: ${{ matrix.integ-config.retry_count || 3 }}
+  #   e2e-test-runner:
```
Comment (david-mcafee) [inline .github/workflows/callable-e2e-tests.yml:26]: Will remove before merging PR.

File: .github/workflows/push-preid-release.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ on:
   push:
     branches:
       # Change this to your branch name where "example-preid" corresponds to the preid you want your changes released on
-      - feat/example-preid-branch/main
+      - gen2-rn-add-tests
```
Comment (david-mcafee) [inline .github/workflows/push-preid-release.yml:12]: Will revert before merging PR.

File: .github/workflows/push-preid-release.yml [hunk 4]
```diff
@@ -25,14 +25,13 @@ jobs:
...
-    needs:
-      - e2e
-      - parse-preid
-    secrets: inherit
-    uses: ./.github/workflows/callable-npm-publish-preid.yml
-    # The preid should be detected from the branch name recommending feat/{PREID}/whatever as branch naming pattern
-    #   if your branch doesn't follow this pattern, you can override it here for your branch.
-    with:
-      preid: ${{ needs.parse-preid.outputs.preid }}
+#   preid-release:
```
Comment (david-mcafee) [inline .github/workflows/push-preid-release.yml:28]: Will revert before merging PR.

**Review by jimblanc — COMMENTED** [review]
LGTM, will approve once test changes reverted

**Review by iartemiev — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Pull request merged**